Add remaining repos to be updated when this workflow is triggered

stefannibrasil opened the pull request:
All of these repos have had the workflow and snippet added to them. This is the final step (I believe) missing to keep our repos in sync with the template.

Unfortunately, for this workflow to work, GitHub seems to need the repo list to be defined explicitly, instead of using `github.repository` or something else more generic.

Using a matrix, we can iterate over the repo list in the steps section. Not sure if there is a better way to do this, though.

### About the repos

I went through the most active repos. I only added the snippet and workflow to the projects that already have the footer/are not forked. Feel free to keep adding the [workflow + snippet](https://github.com/thoughtbot/climate_control/pull/56) to any missing project that you encounter.

sbs-cats committed:
Add remaining repos to be updated when this workflow is triggered

All of these repos have had the workflow and snippet added to them.

Unfortunately, for this workflow to work, GitHub seems to need the repo list to be defined explicitly, instead of using `github.repository` or something else more generic.

Using a matrix, we can iterate over the repo list in the steps section. Not sure if there is a better way to do this, though.

sbs-cats committed:
Update README and footer sections with template

Also, add instructions on how to use the template
when creating a new repo.

stefannibrasil commented on .github/workflows/dynamic-readme-example.yaml:
```diff
@@ -0,0 +1,17 @@
+name: update-templates
```
Added this as an example for folks to copy and paste on the new repo being created.

stefannibrasil commented on README.md:
```diff
@@ -26,16 +35,5 @@ under the terms specified in the [LICENSE] file.
-
-  [community]: https://thoughtbot.com/community?utm_source=github
-  [hire]: https://thoughtbot.com/hire-us?utm_source=github
+<!-- START /templates/footer.md -->
```
I imagine this will work 😎

stefannibrasil commented on README.md.template:
```diff
@@ -22,18 +22,5 @@ under the terms specified in the [LICENSE] file.
 
-## About
-
-$(PROJECT_NAME) is maintained by $(MAINTAINERS).
```
We'll lose these variables for now. I updated lots of repos and didn't see this line being used here. The repos already had their maintainers names alongside the copyright section, so it won't be a huge loss. But it's always something we can add back later.

minaslater commented on .github/workflows/trigger-dynamic-readme-update.yaml:
```diff
@@ -14,7 +14,51 @@ jobs:
     strategy:
       matrix:
-        repository: [thoughtbot/high_voltage]
+        repository:
```
So many of these I have never heard of 😅

minaslater replied:
While I'm usually in the camp of "nothing is impossible", I think the GHAs are not really _that_ smart as to infer which external repo's workflow you want to trigger.

stefannibrasil replied:
Same! :D

> While I'm usually in the camp of "nothing is impossible", I think the GHAs are not really that smart as to infer which external repo's workflow you want to trigger.

yeah, that makes sense. Unless in the repo that has the `workflow_dispatch` key, we could add which workflow path to be listening to updates. So the triggering would work the other way around. I wonder if that's something we could pitch as an idea?

sbs-cats committed:
Document additional step for dynamic README

Besides adding the workflow to the new repo being created,
the new repo path needs to be added to the repo list matrix in the
dispatch workflow. This way, GitHub will propagate the footer
updates to all repos in the list.
